gensim/models/word2vec.py

```diff
@@ -327,9 +328,10 @@ class Word2Vec(utils.SaveLoad):
     compatible with the original word2vec implementation via `save_word2vec_format()` and `load_word2vec_format()`.
 
     """
-    def __init__(self, sentences=None, size=100, alpha=0.025, window=5, min_count=5,
-                 sample=0, seed=1, workers=1, min_alpha=0.0001, sg=1, hs=1, negative=0,
-                 cbow_mean=0, hashfxn=hash, iter=1, null_word=0):
+    def __init__(
+            self, sentences=None, size=100, alpha=0.025, window=5, min_count=5,
+            max_vocab_size=10000000, sample=0, seed=1, workers=1, min_alpha=0.0001,
```

Comment:
I'd prefer a `None` default – it matches prior behavior and avoids correctness surprises like, "word X is missing even though I know it has more than min_count occurences in my corpus!" Meanwhile, choosing to turn it on means you're accepting the compromises it involves.